src/app/components/(imageAnimation)/page.tsx

```diff
@@ -0,0 +1,447 @@
+      {text.split('').map((char: string, i: number) => (
+        <motion.span
+          key={i}
+          className="inline-block"
+          initial={{ opacity: 0, y: 20, scale: 0.8 }}
+          whileInView={{ opacity: 1, y: 0, scale: 1 }}
+          transition={{
+            duration: 0.6,
+            delay: delay + i * 0.04,
+            type: 'spring',
+            stiffness: 100,
+            damping: 10
+          }}
+          viewport={{ once: false, amount: 0.1 }}
+        >
+          {char === ' ' ? '\u00A0' : char}
+        </motion.span>
+      ))}
+    </motion.div>
+  );
+};
+
+// 애니메이션 단어 컴포넌트
+const AnimatedWords = ({ text, delay = 0, className = '' }: { text: string; delay?: number; className?: string }) => {
+  return (
+    <motion.div className={`flex flex-wrap items-center justify-center ${className}`}>
+      {text.split(' ').map((word: string, i: number) => (
+        <motion.span
+          key={i}
+          className="mr-2 inline-block"
+          initial={{ opacity: 0, y: i % 2 === 0 ? 20 : -20, x: i % 3 === 0 ? -10 : i % 3 === 1 ? 10 : 0 }}
+          whileInView={{ opacity: 1, y: 0, x: 0 }}
+          transition={{
+            duration: 0.7,
+            delay: delay + i * 0.07,
+            type: 'spring',
+            stiffness: 100
+          }}
+          viewport={{ once: true, amount: 1 }}
+        >
+          {word}
+        </motion.span>
+      ))}
+    </motion.div>
+  );
+};
+
+// 스크롤 기반 애니메이션 값 추출 함수
+const useScrollAnimation = (
+  scrollYProgress: MotionValue<number>,
+  outputRange: Array<number | string>,
+  inputRange: number[] = [0, 1]
+) => {
+  return useTransform(scrollYProgress, inputRange, outputRange);
+};
+
+export default function ImageAnimation() {
+  const containerRef = useRef(null);
+  const [activeSection, setActiveSection] = useState(0);
+
+  // 전체 스크롤 진행률
+  const { scrollYProgress } = useScroll({
+    target: containerRef,
+    offset: ['start start', 'end end']
+  });
+
+  // 각 섹션의 시작과 끝 범위 정의 (0-1 사이의 값)
+  // 섹션 간에 겹치는 부분 없이 연속적으로 배치
+  const sectionRanges = [
+    [0.05, 0.35], // 첫 번째 섹션 (5%~35%)
+    [0.35, 0.65], // 두 번째 섹션 (35%~65%)
+    [0.7, 0.95] // 세 번째 섹션 (70%~95%)
+  ];
+
+  // 가로 스크롤 효과 - 시작부터 마지막까지 부드러운 이동
+  // 각 섹션의 전환 포인트를 조정하여 애니메이션이 끝난 후에 스크롤되도록 함
+  const x = useTransform(
+    scrollYProgress,
+    [0.05, 0.32, 0.4, 0.67, 0.7, 0.95],
+    ['0vw', '0vw', '-100vw', '-100vw', '-200vw', '-200vw']
+  );
+
+  // 액티브 섹션 추적
+  useEffect(() => {
+    const unsubscribe = scrollYProgress.onChange((value) => {
+      for (let i = 0; i < sectionRanges.length; i++) {
+        if (value >= sectionRanges[i][0] && value <= sectionRanges[i][1]) {
+          setActiveSection(i);
+          break;
+        } else if (value < sectionRanges[0][0]) {
+          setActiveSection(-1); // 인트로 영역
+        } else if (value > sectionRanges[sectionRanges.length - 1][1]) {
+          setActiveSection(sectionRanges.length); // 아웃트로 영역
+        }
+      }
+    });
```

Comment:
`scrollYProgress.onChange`는 이전 버전에서 사용하던 방식이라고 합니다.
`scrollYProgress.on("change", callback)` 방식을 권장한다고 해요.

```suggestion
    const unsubscribe = scrollYProgress.on('change', (value) => {
      for (let i = 0; i < sectionRanges.length; i++) {
        if (value >= sectionRanges[i][0] && value <= sectionRanges[i][1]) {
          setActiveSection(i);
          break;
        } else if (value < sectionRanges[0][0]) {
          setActiveSection(-1); // 인트로 영역
        } else if (value > sectionRanges[sectionRanges.length - 1][1]) {
          setActiveSection(sectionRanges.length); // 아웃트로 영역
        }
      }
    })
```